src/charm.py

```diff
@@ -43,6 +43,7 @@ def __init__(self, *args: typing.Any):
         self.framework.observe(self.on.start, self._on_start)
         self.framework.observe(self.on.config_changed, self._on_config_changed)
         self.framework.observe(self.on.upgrade_charm, self._on_upgrade_charm)
+        self.framework.observe(self.on.update_status, self._on_update_status)
```

Comment:
What your intention here?

Reply:
The `jenkins-agent` service currently doesn't restart when the connection is broken. I added [this commit](d64f0a185ccb5a13384bda608e8422342dda8c74) to automatically restart it with a timeout of 5m.

Since the timeout is 5m I'd need to check with `update_status` if the service cannot be restarted and put the charm in error state.

**The jenkins-agent-k8s** does not do this and just tries to restart infinitely, but I don't think it's a good approach because if multiple restarts doesn't bring the agent back up, then there's an unrecoverable problem and the operator need to be notified.

Reply:
I understand. What would occur if the update interval is shorter that 5m?

Reply:
If an `update-status` event is fired while the service is restarting, the charm will be set to an error state, but if somehow the service managed to restart itself then the status of the charm will be reset to active when the next `update-status` event is fired.